pints/residuals_diagnostics.py

```diff
@@ -11,6 +11,311 @@
 import scipy.special
 
 
+def plot_residuals_binned_autocorrelation(parameters,
+                                          problem,
+                                          thinning=None,
+                                          n_bins=25):
+    r"""
+    Plot the autocorrelation of the residuals within bins.
+
+    Given a time series with observed residuals
+
+    .. math::
+        e_i = y_i - f(t_i; \theta)
+
+    This method divides the vector of residuals into some number of equally
+    sized bins. The lag 1 autocorrelation is calculated for the residuals
+    within each bin. The plot shows the lag 1 autocorrelation in each bin over
+    time.
+
+    This diagnostic is particularly useful for diagnosing time series whose
```

Comment:
I would phrase this as, "This diagnostic is useful for diagnosing time series with noise whose autocorrelation varies over time."